RecoMET/METPUSubtraction/src/PFMETAlgorithmMVA.cc

```diff
@@ -15,41 +15,79 @@
 
 enum MVAType { kBaseline = 0 };
 
-const double Pi=cos(-1);
+const double Pi=std::cos(-1);
 
-namespace
+const std::string PFMETAlgorithmMVA::updateVariableNames(std::string input)
 {
-  const GBRForest* loadMVAfromFile(const edm::FileInPath& inputFileName, const std::string& mvaName)
-  {
-    if ( inputFileName.location()==edm::FileInPath::Unknown ) throw cms::Exception("PFMETAlgorithmMVA::loadMVA") 
-      << " Failed to find File = " << inputFileName << " !!\n";
-    TFile* inputFile = new TFile(inputFileName.fullPath().data());
-  
-    //const GBRForest* mva = dynamic_cast<GBRForest*>(inputFile->Get(mvaName.data())); // CV: dynamic_cast<GBRForest*> fails for some reason ?!
-    const GBRForest* mva = (GBRForest*)inputFile->Get(mvaName.data());
-    if ( !mva )
-      throw cms::Exception("PFMETAlgorithmMVA::loadMVA")
-        << " Failed to load MVA = " << mvaName.data() << " from file = " << inputFileName.fullPath().data() << " !!\n";
-  
-    delete inputFile;
+  if(input=="sumet")     return "particleFlow_SumET";
+  if(input=="npv")       return "nPV";
+  if(input=="pfu")       return "particleFlow_U";
+  if(input=="pfuphi")    return "particleFlow_UPhi";
+  if(input=="tksumet")   return "track_SumET";
+  if(input=="tku")       return "track_U";
+  if(input=="tkuphi")    return "track_UPhi";
+  if(input=="nopusumet") return "noPileUp_SumET";
+  if(input=="nopuu")     return "noPileUp_U";
+  if(input=="nopuuphi")  return "noPileUp_UPhi";
+  if(input=="pusumet")   return "pileUp_SumET";
+  if(input=="pumet")     return "pileUp_MET";
+  if(input=="pumetphi")  return "pileUp_METPhi";
+  if(input=="pucsumet")  return "pileUpCorrected_SumET";
+  if(input=="pucu")      return "pileUpCorrected_U";
+  if(input=="pucuphi")   return "pileUpCorrected_UPhi";
+  if(input=="jetpt1")    return "jet1_pT";
+  if(input=="jeteta1")   return "jet1_eta";
+  if(input=="jetphi1")   return "jet1_Phi";
+  if(input=="jetpt2")    return "jet2_pT";
+  if(input=="jeteta2")   return "jet2_eta";
+  if(input=="jetphi2")   return "jet2_Phi";
+  if(input=="nalljet")   return "nJets";
+  if(input=="njet")      return "numJetsPtGt30";
+  if(input=="uphi_mva")  return "PhiCor_UPhi";
+  if(input=="uphix_mva") return "PhiCor_UPhi";
+  if(input=="ux_mva")    return "RecoilCor_U";
+  return input;
+}
 
-    return mva;
-  }
+const GBRForest* PFMETAlgorithmMVA::loadMVAfromFile(const edm::FileInPath& inputFileName, const std::string& mvaName)
+{
+  if ( inputFileName.location()==edm::FileInPath::Unknown ) throw cms::Exception("PFMETAlgorithmMVA::loadMVA") 
+    << " Failed to find File = " << inputFileName << " !!\n";
+  TFile* inputFile = new TFile(inputFileName.fullPath().data());
```

Comment:
Please use std::unique_ptr\<TFile\> to avoid memory leaks, in particular, because there are throw statements below that would prevent deletion of the object.